tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -98,8 +96,19 @@ public TracerSettings(IConfigurationSource source)
                          new ConcurrentDictionary<string, string>();
 
             // Filter out tags with empty keys or empty values, and trim whitespace
-            GlobalTags = GlobalTags.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
-                                   .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
+            if (GlobalTags.Count > 0)
+            {
+                var globalTags = new Dictionary<string, string>();
```

Comment:
initialize with capacity?

Reply:
done in [9b271d6](https://github.com/DataDog/dd-trace-dotnet/pull/3735/commits/9b271d6330489f7f37df13159a5f41abe6ef8bb8)